### PR TITLE
Clean up and update various items in PETSc/libMesh configure and build scripts

### PR DESCRIPTION
**Pull request by cticenhour**
While working on #23502, there were several out-of-date items with our build/configure scripts that I wanted to update. Putting those items in this PR for later.

### CODE REVIEW

File: scripts/configure_petsc.sh [hunk 1]
```diff
@@ -68,24 +68,21 @@ function configure_petsc()
...
-  MUMPS_ARM_STR=""
   if [[ `uname -p` == "arm" ]] && [[ $(uname) == Darwin ]] && [[ $PETSC_ARCH == arch-moose ]]; then
     echo "INFO: Apple Silicon detected, checking to see if PETSc ARM patches need to be applied..."
     # First check to see if patch marker file exists in PETSC_DIR due to a previous build. If not,
     # perform patch and create patch marker file. If so, skip patch and report.
     patch_file=$(find "$PETSC_DIR" -name '.patched' -print -quit 2>/dev/null)
     if [ -z "$patch_file" ]; then
       # If an HDF5 download is requested, patch PETSc to properly configure it
-      if [ "$HDF5_STR" == "--download-hdf5=1" ]; then
+      if [ "$HDF5_STR" == *"--download-hdf5=1"* ]; then
```
Comment (milljm) [inline scripts/configure_petsc.sh:85]: *neat* :)

**Comment by cticenhour** [comment]
Removed comments from @moosebuild related to not updating the conda YAML files, as I have updated them.

**Comment by cticenhour** [comment]
@milljm Is there anything in this PR that we want to combine with a package change you are making?

**Comment by cticenhour** [comment]
I think I have one more tweak to make, but everything is passing for the moment, so I wanted to check in.

**Review by milljm — DISMISSED** [review]
👍

**Review by lindsayad — APPROVED** [review]
This LGTM

**Pull request merged**